prime-angular-frontend/src/app/modules/adjudication/shared/components/site-registration-container/site-registration-container.component.ts

```diff
@@ -81,6 +82,71 @@ export class SiteRegistrationContainerComponent implements OnInit {
       : this.deleteSite(record.siteId);
   }
 
+  public onApprove(siteId: number) {
+    const data: DialogOptions = {
+      title: 'Approve Site Registration',
+      message: 'Are you sure you want to approve this Registration?',
+      actionText: 'Approve Site Registration',
+      component: NoteComponent
+    };
+
+    let note: string;
+
+    this.busy = this.dialog.open(ConfirmDialogComponent, { data })
+      .afterClosed()
+      .pipe(
+        exhaustMap((result: { output: string }) => {
+          if (result?.output) {
+            note = result.output;
+          }
+
+          return (result)
+            ? of(noop)
+            : EMPTY;
+        }),
+        exhaustMap(() => this.siteResource.approveSite(siteId)),
+        exhaustMap(() =>
+          (note)
+            ? this.siteResource.createSiteRegistrationNote(siteId, note)
+            : of(noop)
+        ),
+      )
+      .subscribe();
+  }
+
+  public onDecline(siteId: number) {
+    const data: DialogOptions = {
+      title: 'Decline Site Registration',
+      message: 'Are you sure you want to Decline this Site Registration?',
+      actionText: 'Decline Site Registration',
+      component: NoteComponent
+    };
+
+    let note: string;
```

Comment:
We could reduce the use of closures and keep the observable pure by piping note through the stream